### PR TITLE
[main][deploy] deployment script for xALPACA.

### PR DESCRIPTION
**Pull request by blackalpacaa**
## Description

- Create deployment script for xAlpacaa contract.

## Why?

- In order to deploy xAlpaca, Grasshouse, AlpacaFeeder to testnet or mainnet.

### CODE REVIEW

**Comment by alpanaca** [comment]
I suggest we put a sequence of contract deployment in the readme.md

Step:
1. Deploy xAlpaca
2. Deploy proxy token
3. Deploy Grasshouse
4. Deploy Feeder

**Review by alpanaca — APPROVED** [review]

File: deploy/exec/alpaca-feeder/deploy/alpaca-feeder.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    POOL_ID,
+    ALPACA_GRASSHOUSE,
+  ])) as AlpacaFeeder;
+  await alpacaFeeder.deployed();
+  console.log(`>> Deployed at ${alpacaFeeder.address}`);
+  console.log("✅ Done");
+
+  console.log(">> Transferring ownership and set okHolders of proxyToken to be alpacaFeeder");
+  const proxyToken = ProxyToken__factory.connect(PROXY_TOKEN, deployer);
+  await proxyToken.setOkHolders([alpacaFeeder.address], true);
```
Comment (sora-alpaca) [inline deploy/exec/alpaca-feeder/deploy/alpaca-feeder.ts:40]: should we add fairlaunch to be okHolders

File: deploy/exec/grasshouse/config/checkpoint.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { DeployFunction } from "hardhat-deploy/types";
+import { ethers, upgrades } from "hardhat";
+import { GrassHouse, GrassHouse__factory } from "../../../../typechain";
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  interface IGrassHouse {
+    SYMBOL: string;
+    GRASSHOUSE_ADDRESS: string;
```
Comment (sora-alpaca) [inline deploy/exec/grasshouse/config/checkpoint.ts:9]: why uppercase

File: deploy/exec/proxy-token/deploy/proxy-token.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    ░██╗░░░░░░░██╗░█████╗░██████╗░███╗░░██╗██╗███╗░░██╗░██████╗░
+    ░██║░░██╗░░██║██╔══██╗██╔══██╗████╗░██║██║████╗░██║██╔════╝░
+    ░╚██╗████╗██╔╝███████║██████╔╝██╔██╗██║██║██╔██╗██║██║░░██╗░
+    ░░████╔═████║░██╔══██║██╔══██╗██║╚████║██║██║╚████║██║░░╚██╗
+    ░░╚██╔╝░╚██╔╝░██║░░██║██║░░██║██║░╚███║██║██║░╚███║╚██████╔╝
+    ░░░╚═╝░░░╚═╝░░╚═╝░░╚═╝╚═╝░░╚═╝╚═╝░░╚══╝╚═╝╚═╝░░╚══╝░╚═════╝░
+    Check all variables below before execute the deployment script
+    */
+  const SYMBOL = "ProxyToken";
+  const TIMELOCK = "0x2D5408f2287BF9F9B05404794459a846651D0a59";
```
Comment (sora-alpaca) [inline deploy/exec/proxy-token/deploy/proxy-token.ts:18]: should we do this as constant

File: deploy/exec/xalpaca/deploy/xalpaca.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  /*
+    ░██╗░░░░░░░██╗░█████╗░██████╗░███╗░░██╗██╗███╗░░██╗░██████╗░
+    ░██║░░██╗░░██║██╔══██╗██╔══██╗████╗░██║██║████╗░██║██╔════╝░
+    ░╚██╗████╗██╔╝███████║██████╔╝██╔██╗██║██║██╔██╗██║██║░░██╗░
+    ░░████╔═████║░██╔══██║██╔══██╗██║╚████║██║██║╚████║██║░░╚██╗
+    ░░╚██╔╝░╚██╔╝░██║░░██║██║░░██║██║░╚███║██║██║░╚███║╚██████╔╝
+    ░░░╚═╝░░░╚═╝░░╚═╝░░╚═╝╚═╝░░╚═╝╚═╝░░╚══╝╚═╝╚═╝░░╚══╝░╚═════╝░
+    Check all variables below before execute the deployment script
+*/
+  const ALPACA_TOKEN = ""; // Address of alpaca token
```
Comment (sora-alpaca) [inline deploy/exec/xalpaca/deploy/xalpaca.ts:16]: constant

**Review by sora-alpaca — CHANGES_REQUESTED** [review]

File: deploy/exec/alpaca-feeder/deploy/alpaca-feeder.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  /*
+    ░██╗░░░░░░░██╗░█████╗░██████╗░███╗░░██╗██╗███╗░░██╗░██████╗░
+    ░██║░░██╗░░██║██╔══██╗██╔══██╗████╗░██║██║████╗░██║██╔════╝░
+    ░╚██╗████╗██╔╝███████║██████╔╝██╔██╗██║██║██╔██╗██║██║░░██╗░
+    ░░████╔═████║░██╔══██║██╔══██╗██║╚████║██║██║╚████║██║░░╚██╗
+    ░░╚██╔╝░╚██╔╝░██║░░██║██║░░██║██║░╚███║██║██║░╚███║╚██████╔╝
+    ░░░╚═╝░░░╚═╝░░╚═╝░░╚═╝╚═╝░░╚═╝╚═╝░░╚══╝╚═╝╚═╝░░╚══╝░╚═════╝░
+    Check all variables below before execute the deployment script
+    */
+  const ALPACA_TOKEN = "";
```
Comment (spicysquid168) [inline deploy/exec/alpaca-feeder/deploy/alpaca-feeder.ts:17]: Please move all known variables to json files instead i.e. ALPACA_TOKEN, FAIR_LAUNCH, PROXY_TOKEN

File: deploy/exec/proxy-token/deploy/proxy-token.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    ░██╗░░░░░░░██╗░█████╗░██████╗░███╗░░██╗██╗███╗░░██╗░██████╗░
+    ░██║░░██╗░░██║██╔══██╗██╔══██╗████╗░██║██║████╗░██║██╔════╝░
+    ░╚██╗████╗██╔╝███████║██████╔╝██╔██╗██║██║██╔██╗██║██║░░██╗░
+    ░░████╔═████║░██╔══██║██╔══██╗██║╚████║██║██║╚████║██║░░╚██╗
+    ░░╚██╔╝░╚██╔╝░██║░░██║██║░░██║██║░╚███║██║██║░╚███║╚██████╔╝
+    ░░░╚═╝░░░╚═╝░░╚═╝░░╚═╝╚═╝░░╚═╝╚═╝░░╚══╝╚═╝╚═╝░░╚══╝░╚═════╝░
+    Check all variables below before execute the deployment script
+    */
+  const SYMBOL = "ProxyToken";
+  const TIMELOCK = ""; // Address of TIMELOCK
```
Comment (spicysquid168) [inline deploy/exec/proxy-token/deploy/proxy-token.ts:18]: This is known contract, hence better read it from json file

File: deploy/exec/xalpaca/deploy/xalpaca.ts (hunk 4 above)
Comment (spicysquid168) [inline deploy/exec/xalpaca/deploy/xalpaca.ts:16]: read from json file

**Review by spicysquid168 — CHANGES_REQUESTED** [review]

**Review by adirutwn — APPROVED** [review]

**Review by spicysquid168 — APPROVED** [review]

**Review by sora-alpaca — APPROVED** [review]

**Pull request merged**